custom/src/main/java/org/kiwix/kiwixmobile/custom/download/CustomDownloadFragment.kt

```diff
@@ -76,6 +96,14 @@ class CustomDownloadActivity : BaseActivity() {
     compositeDisposable.clear()
   }
 
+  override fun onBackPressed(activity: AppCompatActivity): FragmentActivityExtensions.Super {
+    pressBackCallback = 1
+    downloadViewModel.actions.offer(PressedBack)
```

Comment:
without running the code this seems very odd to me, pressing back will trigger a side effect to navigate to CustomReader? seeing as you call `super.onBackPressed()` that will return a ShouldCall then we would call the super implemenetation in CoreMainActivity

```
  override fun onBackPressed() {
    if (navigationDrawerIsOpen()) {
      closeNavigationDrawer()
      return
    }
    if (activeFragments().filterIsInstance<FragmentActivityExtensions>().isEmpty()) {
      return super.onBackPressed()
    }
    activeFragments().filterIsInstance<FragmentActivityExtensions>().forEach {
      if (it.onBackPressed(this) == FragmentActivityExtensions.Super.ShouldCall) {
        super.onBackPressed()
      }
    }
  }
```
and that would just exit the app if I've reasoned it through correctly. So the behaviour is what we want ie you cannot go back to Reader from Download but there is a lot of noise surrounding it. Deleting this method should be sufficient

Reply:
If we don't add our custom back handling here then, it will never exit from the download screen no matter how many times on presses back button, so we'll have to provide a custom definition here

Reply:
https://github.com/kiwix/kiwix-android/pull/2531 here I made this pull request to demonstrate the desired behaviour.

Reply:
In the PR you mentioned above, there is no clue how back press is being handled after the deletion of the corresponding method, also you haven't manipulated stack there which clarifies stuff. When I made the changes demanded by you in current PR, It is now at the old state, unable to exit from download screen, pressing back on which sends to customReader, which redirects back to the download screen, popping the backstack works in initial two runs only(i.e., doing so makes us able to directly exit from download screen), and in further runs it fails and throws crashes

Reply:
Yeah, I understood that PR(you may cross-question me to verify), there, on navigating to destinations we've cleared the backstack and therefore when back press is triggered, it simply exits the application. I did the same thing via navigating to download screen and simultaneously clear the backstack, but this doesn't seem to work, as when back press is triggered, it sends back to reader and then the loop continues.

Just as an FYI, this behavior pre-exists on `develop` for _custom_ and isn't introduced in this PR;

Anyways, but why does that ideology not work when I implement it here in the current scenario(about the one that I mentioned few lines back)?

Reply:
If you use the `popUpTo/Inclusive` for navigating to/from CustomDownload/CustomReader and remove onBackPressed overload here and follow the advice I have given it will work. 

Why would this scenario be ocurring?
> add our custom back handling here then, it will never exit from the download screen no matter how many times on presses back button
Because if the previous destination is on the backstack when you go back to it it will immediately send you to download again because that is what that screen does, if it can't find a file it boots you to download. Or that is what I guess is happening